python_modules/libraries/dagster-dg/dagster_dg/cli/plus.py

```diff
@@ -142,3 +151,169 @@ def pull_env_command(**global_options: object) -> None:
             click.echo(
                 f"Environment variables not found for projects: {', '.join(projects_without_secrets)}"
             )
+
+
+class Scope(str, Enum):
+    LOCAL = "local"
+    BRANCH = "branch"
+    FULL = "full"
+
+
+def _secret_is_global(secret: Mapping[str, Any]) -> bool:
+    return not secret["locationNames"]
+
+
+def _get_secret_scopes(secret: Mapping[str, Any]) -> set[Scope]:
+    return {
+        s
+        for s in {
+            Scope.FULL if secret["fullDeploymentScope"] else None,
+            Scope.BRANCH if secret["allBranchDeploymentsScope"] else None,
+            Scope.LOCAL if secret["localDeploymentScope"] else None,
+        }
+        if s is not None
+    }
+
+
+@plus_env_group.command(name="add", cls=DgClickCommand)
+@click.argument("env_name")
+@click.argument("env_value", type=click.STRING, required=False)
+@click.option(
+    "--from-local-env",
+    is_flag=True,
+    help="Pull the environment variable value from your shell environment or project .env file.",
+)
+@click.option(
+    "--scope",
+    type=click.Choice(["full", "branch", "local"]),
+    multiple=True,
+    help="The deployment scope to set the environment variable in. Defaults to all scopes.",
+)
+@click.option(
+    "--global",
+    "global_",
+    is_flag=True,
+    help="Whether to set the environment variable at the deployment level, for all locations.",
+)
+@dg_global_options
+@cli_telemetry_wrapper
+def add_env_command(
+    env_name: str,
+    env_value: str,
+    scope: list[str],
+    global_: bool,
+    from_local_env: bool,
+    **global_options: object,
+) -> None:
+    """Set an environment variable in Dagster Plus."""
+    if not env_value and not from_local_env:
+        raise click.UsageError(
+            "Environment variable value is required. You can either directly provide this value or use the --from-local-env flag to pull the value from your shell environment or project .env file."
+        )
+    if env_value and from_local_env:
+        raise click.UsageError(
+            "Environment variable value and --from-local-env cannot both be provided."
+        )
+
+    cli_config = normalize_cli_config(global_options, click.get_current_context())
+
+    dg_context = DgContext.for_project_environment(Path.cwd(), cli_config)
```

Comment:
it seems to like you should be able to run this outside of a project context (at least in the global case, which I believe should be the default case)

Reply:
Updated to support on the workspace level, and added tests.